docs/docs-content/clusters/public-cloud/azure/required-permissions.md

```diff
@@ -0,0 +1,366 @@
+| Microsoft.Network/privateDnsZones/A/delete           | Subscription |
+| Microsoft.Network/privateDnsZones/A/read             | Subscription |
+| Microsoft.Network/privateDnsZones/A/write            | Subscription |
+| Microsoft.Network/privateDnsZones/delete             | Subscription |
+| Microsoft.Network/privateDnsZones/read               | Subscription |
+| Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete | Subscription |
+| Microsoft.Network/privateDnsZones/virtualNetworkLinks/read | Subscription |
+| Microsoft.Network/privateDnsZones/virtualNetworkLinks/write | Subscription |
+| Microsoft.Network/privateDnsZones/write              | Subscription |
+| Microsoft.Network/publicIPAddresses/delete           | Subscription |
+| Microsoft.Network/publicIPAddresses/join/action      | Subscription |
+| Microsoft.Network/publicIPAddresses/read             | Subscription |
+| Microsoft.Network/publicIPAddresses/write            | Subscription |
+| Microsoft.Network/routeTables/delete                 | Subscription |
+| Microsoft.Network/routeTables/read                   | Subscription |
+| Microsoft.Network/routeTables/write                  | Subscription |
+| Microsoft.Network/virtualNetworks/delete             | Subscription |
+| Microsoft.Network/virtualNetworks/join/action        | Subscription |
+| Microsoft.Network/virtualNetworks/join/action        | Subscription |
+| Microsoft.Network/virtualNetworks/joinLoadBalancer/action | Subscription |
+| Microsoft.Network/virtualNetworks/peer/action         | Subscription |
+| Microsoft.Network/virtualNetworks/read               | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/delete     | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/join/action| Subscription |
+| Microsoft.Network/virtualNetworks/subnets/joinLoadBalancer/action | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/read       | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/virtualMachines/read | Subscription |
+| Microsoft.Network/virtualNetworks/subnets/write      | Subscription |
+| Microsoft.Network/virtualNetworks/virtualMachines/read | Subscription |
+| Microsoft.Network/virtualNetworks/virtualNetworkPeerings/delete | Subscription |
+| Microsoft.Network/virtualNetworks/virtualNetworkPeerings/read | Subscription |
+| Microsoft.Network/virtualNetworks/virtualNetworkPeerings/write | Subscription |
+| Microsoft.Network/virtualNetworks/write              | Subscription |
+| Microsoft.Resources/subscriptions/resourceGroups/read | Subscription |
+| Microsoft.Storage/storageAccounts/blobServices/containers/read | Subscription |
+| Microsoft.Storage/storageAccounts/blobServices/containers/write | Subscription |
+| Microsoft.Storage/storageAccounts/blobServices/listKeys/action | Subscription |
+| Microsoft.Storage/storageAccounts/read                | Subscription |
+| Microsoft.Storage/storageAccounts/write               | Subscription |
+
+
+
+
+## Validator
```

Comment:
This feels out of context here. Static and dynamic placement are types of management in Palette, and Validator is a tool. It might be the case that I don't understand the technical side of it, but purely from the structure perspective, I'd treat Validator as different from static and dynamic placement.

Reply:
I wanted to jump in and add some context.

The Validator tool is something we created to help customers deploy Palette. Its purpose is to help our customers ensure that they've followed the instructions under **Static Placement** and **Dynamic Placement** correctly - that they've added the permissions required to use Palette. For Validator to work, it itself needs permissions. They're kind of "meta" permissions - permissions that allow a service principal to check permissions on other service principals.

You'e right that the act of using Validator is something completely different from the act of using Palette. We were trying to figure out the best place to tell our users which Azure permissions they need in order to use Validator.